src/resources/IPXInterfaces/IPXInterface.model.ts

```diff
@@ -50,7 +50,103 @@ export interface IPXResultTemplate extends HostedInterfaceResultTemplate {
 }
 
 export interface IPXButton {
-    label: string;
+    /**
+     * The label appearing on the IPX button.
+     */
+    label?: string;
+
+    /**
+     * The icon appearing on the IPX button when the IPX is open.
+     */
+    closeIcon?: string;
+
+    /**
+     * The icon appearing on the IPX button when the IPX is close.
+     */
+    openIcon?: string;
+}
+
+export interface IPXContainer {
+    /**
+     * Whether the IPX is using a target selector.
+     */
+    usesTargetSelector?: boolean;
+
+    /**
+     * The IPX button information.
+     */
+    button?: IPXButton;
+
+    /**
+     * The CSS selector used to target the host component of the IPX when usesTargetSelector is true.
+     */
+    targetSelector?: string;
+}
```

Comment:
Hmm, but if I load an IPX that has `usesTargetSelector: false`, it would interpret it as an AutomaticIPXContainer and I think it would lose the `targetSelector ` string on the next save (which is a bit of a pain if the user wants to reenable it after that)